### PR TITLE
Resolves action and function OperationIds for PowerShell style

### PR DESCRIPTION
**Pull request by irvinesunday**
Closes https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/543

**Summary:**
PowerShell had breaking changes due to this fix: https://github.com/microsoft/OpenAPI.NET.OData/pull/98. 
This fix was included in the `Microsoft.OpenApi.OData` package `v1.0.7`. This caused cmdlets names generated from _action_ and _function_ operations to change due to changed _OperationIds_. The target library version had to be reverted (to `v1.0.6`) to reverse this breaking change. There is need however, for bumping up the version while still continuing to provide the previous experience to PowerShell. In the OpenApiService in DevX API, we can modify certain aspects of the output OpenAPI doc. before serving the payload back to the requesting client. For the PowerShell style, we need to have a way of resolving _action_/_functions_ _OperationIds_ generated by the `Microsoft.OpenApi.OData` package. This PR aims to achieve that.

**Proposes:**
- Updating  `Microsoft.OpenApi.OData` from `v1.0.6` to `v1.0.7`
- Adding support for resolving _action_ and _function_ OperationIds by reverting their signatures to how they were being defined in the package `Microsoft.OpenApi.OData` `v1.0.6` and below.
- Adding and updating tests to validate the above.
- Minor code cleanup in the test file.

An example of an _action_ _OperationId_ generated from `v1.0.7` --> `communications.calls.call_keepAlive`
The same _OperationId_ generated from `v1.0.6` (and as expected by PowerShell so as to maintain the cmdlet names) --> `communications.calls_keepAlive`

### CODE REVIEW

File: OpenAPIService/PowershellFormatter.cs [hunk 1]
```diff
@@ -47,6 +50,23 @@ public override void Visit(OpenApiOperation operation)
         {
             var operationId = operation.OperationId;
 
+            if (operation.Extensions.TryGetValue("x-ms-docs-operation-type",
+                                                  out var value) && value != null)
+            {
+                var operationType = ((OpenApiString)value).Value;
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:56]:
```suggestion
                var operationType = (value as OpenApiString)?.Value;
```

File: OpenAPIService/PowershellFormatter.cs [hunk 2]
```diff
@@ -47,6 +50,23 @@ public override void Visit(OpenApiOperation operation)
...
+
+                if (operationType.Equals("action", StringComparison.OrdinalIgnoreCase) ||
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:58]:
```suggestion
                if ("action".Equals(operationType, StringComparison.OrdinalIgnoreCase) ||
```
this style protects you against operationTypeBeing null

File: OpenAPIService/PowershellFormatter.cs [hunk 3]
```diff
@@ -47,6 +50,23 @@ public override void Visit(OpenApiOperation operation)
...
+                    operationType.Equals("function", StringComparison.OrdinalIgnoreCase))
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:59]:
```suggestion
                    "function".Equals(operationType, StringComparison.OrdinalIgnoreCase))
```

File: OpenAPIService/PowershellFormatter.cs [hunk 4]
```diff
@@ -63,9 +83,47 @@ public override void Visit(OpenApiSchema schema)
...
+        /// <example>
+        /// Default OperationId --> communications.calls.call_keepAlive
+        /// Resolved OperationId --> communications.calls_keepAlive
+        /// </example>
+        /// <param name="operation">The target OpenAPI operation.</param>
+        /// <returns>The resolved OperationId.</returns>
+        private string ResolveActionFunctionOperationId(OpenApiOperation operation)
+        {
+            var operationId = operation.OperationId;
+            var segments = operationId.Split('.', (char)StringSplitOptions.RemoveEmptyEntries).ToList();
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:107]:
```suggestion
            var segments = operationId.Split(new char[] {'.'}, StringSplitOptions.RemoveEmptyEntries).ToList();
```

File: OpenAPIService/PowershellFormatter.cs [hunk 5]
```diff
@@ -63,9 +83,47 @@ public override void Visit(OpenApiSchema schema)
...
+
+            foreach (var parameter in operation.Parameters)
+            {
+                // Get the ODataKeySegment value.
+                // Paths containing the above segment are
+                // the ones needing to be resolved.
+                if (parameter.Extensions.TryGetValue("x-ms-docs-key-type",
+                                                     out var value) && value != null)
+                {
+                    var keyType = ((OpenApiString)value).Value;
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:117]:
```suggestion
                    var keyType = (value as OpenApiString)?.Value;
```

File: OpenAPIService/PowershellFormatter.cs [hunk 6]
```diff
@@ -63,9 +83,47 @@ public override void Visit(OpenApiSchema schema)
...
+
+                    if (operationId.Contains(keyType))
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:119]:
```suggestion
                    if (!string.IsNullOrEmpty(keyType) && operationId.Contains(keyType))
```

**Review by baywet — CHANGES_REQUESTED** [review]

File: OpenAPIService/PowershellFormatter.cs (hunk 2 above)
Comment (irvinesunday) [reply OpenAPIService/PowershellFormatter.cs:58]: Interesting...

**Review by baywet — APPROVED** [review]

**Review by peombwa — APPROVED** [review]

**Pull request merged**